subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/BasicJavaCompilerIntegrationSpec.groovy

```diff
@@ -148,7 +148,16 @@ public class FxApp extends Application {
         given:
         goodCode()
         buildFile << """
+java.targetCompatibility = JavaVersion.VERSION_1_7 // this will be ignored when compiling, but used for the TargetJvmVersion attribute
 compileJava.options.compilerArgs.addAll(['--release', '8'])
+compileJava {
+    doFirst {
+        assert configurations.apiElements.attributes.getAttribute(TargetJvmVersion.TARGET_JVM_VERSION_ATTRIBUTE) == 7
```

Comment:
Is it not worth it to check that a `--release` argument has been provided and then use it?
It would close one more loophole.

Reply:
Users should use the property now. Doing a special handling of the '--release' arg in the compile task, was just added to make it work somehow before we introduced the `release` property. But I won't make the code in other places more complex by doing the special handling there as well. `compilerArgs` is a low level API to do things which Gradle does not yet support. I think we should keep it that way.